src/orion/client/experiment.py

```diff
@@ -668,6 +668,7 @@ def workon(
         max_broken=None,
         trial_arg=None,
         on_error=None,
+        prepare_trial=None,
```

Comment:
I'm curious: In what kind of scenario would the user need/want to pass a different callable than the default that we already use?

Also:
- if we want to make it possible to pass a different function, its expected signature should be described here (preferably using a type hint rather than a docstring).
- Wouldn't it be more accurate to set the default value here to `prepare_trial_working_dir`, rather than `None`?

  ```suggestion
          prepare_trial: Callable[[ExperimentClient, Trial], None]=prepare_trial_working_dir,
  ```

Reply:
> I'm curious: In what kind of scenario would the user need/want to pass a different callable than the default that we already use?

I don't know honestly. If they need to prepare something before the worker starts doing the work...

> if we want to make it possible to pass a different function, its expected signature should be described here (preferably using a type hint rather than a docstring).

Indeed

> Wouldn't it be more accurate to set the default value here to prepare_trial_working_dir, rather than None?

It would be more explicit. I was trying to avoid adding a reference to runner's default inside workon, but the advantage of making it more explicit is better in retrospect.

Reply:
If I want to type the other arguments, what would be the correct Callable for `fct` since it takes `**kwargs`?

Reply:
Fixed in c357931cec3b672251173861c937ef6f117f28a9